### PR TITLE
fix(app): handle question response races clearly

### LINKED ISSUE #837: [Bug] Question dock can show [object Object] after answering
## What happened?

When the assistant asks a question through the question dock, choosing an option can show a toast with:

> 请求失败
> [object Object]

The observed session export shows the answer was actually accepted and the assistant continued with the chosen action. The user-visible failure is therefore misleading: the app reports a raw object instead of a readable response-state message.

Affected observed case:

- Question: `怎么处理这个 perf CI？`
- Selected answer: `重跑 failed job`
- Session export: `docs/debug-session-log/pawwork-session-nimble-engine-2026-05-21-14-47-54-question-failed.json`
- App version in export: `0.0.0-prod-202605201748`

## Which area seems affected?

App flow or product behavior; UI or design system; Model harness, prompts, tools, or session mechanics.

## How much does this affect you?

Breaks an important workflow. The question dock is a decision point; showing `[object Object]` makes it look like the answer failed even when it may have succeeded.

## Steps to reproduce

1. Open a session where the assistant calls the `question` tool.
2. Choose an option in the question dock and submit.
3. In the observed failure, a toast appears: `请求失败 [object Object]`.
4. The session export still shows the question tool completed with the selected answer and the assistant continued.

Potentially related reproduction path: submit the same question response again while the external-result entry has already been resolved, or while the dock remains visible before SSE/global sync removes it.

## What did you expect to happen?

- If the answer is accepted, the dock should close or become non-interactive without showing a failure toast.
- If the answer was already accepted elsewhere or by an earlier click, the app should treat that as an idempotent completion, not a user-facing failure.
- If the response really is stale/invalid, the toast should use readable copy, never `[object Object]`.

## PawWork version

`0.0.0-prod-202605201748`

## OS version

macOS `25.5.0` from the session export.

## Can you reproduce it again?

Sometimes / observed from session export and screenshot.

## Diagnostics

Evidence from the session export:

- The question tool part is completed and contains the selected answer:
  - `state.output`: `User has answered your questions: "怎么处理这个 perf CI？"="重跑 failed job"...`
- The assistant then executed the chosen action:
  - `gh run rerun 26230861270 --failed`
  - exit `0`

Likely involved code paths:

- `packages/app/src/pages/session/composer/session-question-dock.tsx`
- `packages/sdk/js/src/v2/gen/client/client.gen.ts`
- `packages/opencode/src/server/instance/session.ts`
- `packages/opencode/src/tool/external-result.ts`

### PR DESCRIPTION
**Pull request by Astro-Han**
## Summary

- Keep the question dock open on failed `/session/:id/tool/respond` attempts so users can correct and retry without losing drafts.
- Normalize question response errors locally so plain object failures never surface as `[object Object]`, including plain body `no_pending_tool_call` and decoder-detail errors.
- Treat `already_resolved` as idempotent completion only when it matches a request that this dock already submitted, and keep the stale local dock reactively non-interactive after pending/confirmed completion until sync removes it.

## Why

Issue #837 reported that answering a question dock could show unreadable `[object Object]` errors and close the dock before users could fix decoder/payload failures. The fix keeps retryable failures interactive while preserving idempotent completion for the same locally submitted request.

## Related Issue

Fixes #837

## Human Review Status

Pending

## Review Focus

- Confirm `already_resolved` only completes after the same local question request was submitted.
- Confirm plain body `no_pending_tool_call` maps to stale-session copy, and structured decoder errors keep readable details without `[object Object]`.
- Confirm pending and post-success waiting-for-sync states block duplicate submit/dismiss/selection interactions and now drive reactive disabled state.

## Risk Notes

Skipped checklist items:
- Manual visible UI/copy check: not run locally; this change is covered by focused unit tests, browser-reactivity check, and typecheck, but it does alter error-copy routing.
- Platform impact: no platform, packaging, updater, signing, path, shell, or permission surface touched.
- Docs/release/dependencies/etc.: none touched.

## How To Verify

```text
Focused question dock tests: 15 passed / 0 failed
Command: bun test --preload ./happydom.ts src/pages/session/composer/session-question-dock.test.ts

App typecheck: passed
Command: bun run typecheck

Diff check: passed with no whitespace errors
Command: git diff --check
```

## Screenshots or Recordings

Not provided. No screenshot/recording was captured for this copy/error-state change.

## Checklist

> **How to use this checklist:**
>
> - Tick a box by replacing `[ ]` with `[x]`. Do not edit, add, or remove items.
> - The bot-applied label items can only be honestly ticked AFTER the PR is opened and the labeler / priority-triage bots have run — return to the PR description and tick them then.
> - Most items are required. The few that are conditional are explicitly marked **(conditional)**; for those, leave unticked if they truly do not apply and explain why in Risk Notes. All other items must be ticked before requesting human review.

- [x] **Type label** — this PR carries exactly one of `bug`, `enhancement`, `task`, `documentation`. Type labels are author-added; the labeler bot does NOT assign them. Add the label in the GitHub UI, then tick this.
- [x] **Routing labels** — this PR carries at least one of `app`, `ui`, `platform`, `harness`, `ci`. The labeler bot assigns these on PR open based on changed paths. Confirm the bot's choice (or override if wrong), then tick this.
- [x] **Priority label** — this PR carries exactly one of `P0`, `P1`, `P2`, `P3`. The priority-triage bot suggests one on PR open. Confirm or override, then tick this.
- [x] Human Review Status above is set to `Pending`, `Approved by @<reviewer>`, or `Not required: <reason>` (default is `Pending`; "not required" is restricted to bot-authored low-risk PRs).
- [x] I linked the related issue, or stated in Summary why there is no issue.
- [x] I described the review focus and any meaningful risks.
- [x] I replaced the example block in How To Verify with the real verification steps and the key result for each.
- [x] I did not introduce unrelated refactors, dependencies, generated files, or file changes beyond the stated scope.
- [ ] **(conditional)** I manually checked visible UI or copy changes when needed, with screenshots or recordings. Leave unticked only if no visible UI or copy changed.
- [ ] **(conditional)** I considered macOS and Windows impact for platform, packaging, updater, signing, paths, shell, or permissions changes. Leave unticked only if no platform/packaging surface was touched.
- [ ] **(conditional)** I called out docs, release notes, dependencies, permissions, credentials, deletion behavior, generated content, or local file changes when relevant. Leave unticked only if none of those surfaces was touched.
- [x] I reviewed the final diff for unrelated changes and suspicious dependency changes.
- [x] I am targeting `dev`, and my PR title and commit messages use Conventional Commits in English.

## Release Notes

* **Bug Fixes**
  * Question dock now prevents duplicate submissions while awaiting responses by disabling controls during request processing.
  * Improved error handling for tool responses with better detection of stale sessions and invalid requests, providing clearer user feedback.

* **Tests**
  * Added end-to-end and unit tests to verify question dock prevents overlapping submissions and handles edge cases correctly.

[Review Change Stack](https://app.coderabbit.ai/change-stack/Astro-Han/pawwork/pull/839?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: packages/app/src/pages/session/composer/session-question-dock.tsx [hunk 1]
```diff
@@ -207,45 +291,60 @@ export const SessionQuestionDock: Component<{ request: DockQuestionRequest; onSu
...
-      return
+      locallySubmitted = undefined
+      return "failed"
     }
-    const message = err instanceof Error ? err.message : String(err)
-    showToast({ title: language.t("common.requestFailed"), description: message })
+    showToast({
+      title: language.t("common.requestFailed"),
+      description: normalized.detail || language.t("session.question.error.invalidPayload"),
+    })
```
[the remark this answers is not in the document]
Comment (Astro-Han) [reply packages/app/src/pages/session/composer/session-question-dock.tsx:386]: Fixed in dfd9c03ea2. The catch-all question response error path no longer falls back to invalid-payload copy; API-provided details are still surfaced when present, otherwise the toast uses a generic question submission failure message.

File: packages/app/e2e/session/session-composer-dock.spec.ts [hunk 2]
```diff
@@ -738,6 +738,74 @@ test("blocked question flow supports escape dismiss", async ({ page, llm, projec
...
+          await expect(customInput).toBeDisabled()
+          await expect(firstOption).toBeDisabled()
+          await expect(submit).toBeDisabled()
+
+          await dock.evaluate((el) => {
+            el.dispatchEvent(new KeyboardEvent("keydown", { key: "Escape", bubbles: true, cancelable: true }))
+          })
+          await submit.evaluate((button: HTMLButtonElement) => button.click())
+          await firstOption.evaluate((button: HTMLButtonElement) => button.click())
+          await expect.poll(() => responseCalls, { timeout: 1_000 }).toBe(1)
```
[the remark this answers is not in the document]
Comment (Astro-Han) [reply packages/app/e2e/session/session-composer-dock.spec.ts]: Fixed in dfd9c03ea2. The E2E now arms a waitForRequest watcher after the first pending response, forces the raw DOM interactions, and asserts no second /tool/respond request was observed before checking responseCalls is still 1.

File: packages/app/src/pages/session/composer/session-question-dock.tsx [hunk 3]
```diff
@@ -207,45 +331,67 @@ export const SessionQuestionDock: Component<{ request: DockQuestionRequest; onSu
...
+      responseGuard.fail(props.request.id)
+      locallySubmitted = undefined
+      return "failed"
     }
-    const message = err instanceof Error ? err.message : String(err)
-    showToast({ title: language.t("common.requestFailed"), description: message })
+    showToast({
+      title: language.t("common.requestFailed"),
+      description: normalized.detail || language.t("session.question.error.invalidPayload"),
+    })
```
[the remark this answers is not in the document]
Comment (Astro-Han) [reply packages/app/src/pages/session/composer/session-question-dock.tsx:386]: Fixed in dfd9c03ea2. Unknown question submission failures now fall back to session.question.error.unknown, with EN/ZH translations added; invalid_payload still uses the payload-specific copy.
